CMakeLists.txt

```diff
@@ -27,6 +27,23 @@
 cmake_minimum_required(VERSION 2.8.11)
 project(opae)
 
+############################################################################
+## Get git hash info, if available #########################################
+############################################################################
+find_program(GIT_EXECUTABLE git)
+if(EXISTS ${GIT_EXECUTABLE})
+  execute_process(COMMAND ${GIT_EXECUTABLE} log -1 --format=%h
+    WORKING_DIRECTORY ${CMAKE_SOURCE_DIR}
+    OUTPUT_VARIABLE GIT_COMMIT_HASH
+    RESULT_VARIABLE GIT_LOG_RESULT
+    OUTPUT_STRIP_TRAILING_WHITESPACE)
+    if(NOT ${GIT_LOG_RESULT} EQUAL 0)
+      set(GIT_COMMIT_HASH unknown)
+    endif(NOT ${GIT_LOG_RESULT} EQUAL 0)
```

Comment:
What do we see when we do a `git archive`? Should we have this fall back to using something like the `id` you had in another PR?

Reply:
Yes, we should. I'm going to add that in another PR.